uploadTest

```diff
@@ -0,0 +1 @@
+test
```

Comment:
Regrettably, it appears that file ctime/mtime has an effect of the encoded output, making a reliable mock impossible with this approach.  That said, the ctime will be entirely different when run on a smoker for these files too;  This would explain why travis failed this PR.

The only other option for testing is to pass the zipped/base64'd string directly to upload_file.  I consider this an acceptable compromise, and will re-factor the upload_file method to optionally accept this sort of data.